File: Rust/BasicAlgorithms/src/PrintSum.rs
pub fn PrintSum_ForIn() {
    let mut sum = 0;
    for i in 1..256 {
        sum += i;
        println!("New number: {}, Sum: {}", i, sum);
    }
}
pub fn PrintSum_While() {
    let mut sum = 0;
    let mut i = 0;
    while i < 256 {
        sum += i;
        println!("New number: {}, Sum: {}", i, sum);
        i += 1;
    }
}

File: C/BasicAlgorithms/PrintSum.c
#include "basic_algorithms.h"

// Print numbers from 0 to 255 along with the cumulative sum
void print_sum_0_to_255(void) {
    long sum = 0;
    for (int i = 0; i <= 255; i++) {
        sum += i;
        printf("New number: %d Sum: %ld\n", i, sum);
    }
}
